homeassistant/components/evohome/coordinator.py

```diff
@@ -5,85 +5,94 @@
 from collections.abc import Awaitable
 from datetime import timedelta
 import logging
-from typing import Any
+from typing import TYPE_CHECKING, Any
 
 import evohomeasync as ev1
-from evohomeasync.schema import SZ_ID, SZ_SESSION_ID, SZ_TEMP
+from evohomeasync.schema import SZ_ID, SZ_TEMP
 import evohomeasync2 as evo
+from evohomeasync2.schema.const import (
+    SZ_GATEWAY_ID,
+    SZ_GATEWAY_INFO,
+    SZ_LOCATION_ID,
+    SZ_LOCATION_INFO,
+    SZ_TIME_ZONE,
+)
 
-from homeassistant.const import CONF_USERNAME
-from homeassistant.core import HomeAssistant
 from homeassistant.helpers.dispatcher import async_dispatcher_send
 from homeassistant.helpers.event import async_call_later
-from homeassistant.helpers.storage import Store
-from homeassistant.helpers.typing import ConfigType
-
-from .const import (
-    ACCESS_TOKEN,
-    ACCESS_TOKEN_EXPIRES,
-    CONF_LOCATION_IDX,
-    DOMAIN,
-    GWS,
-    REFRESH_TOKEN,
-    TCS,
-    USER_DATA,
-    UTC_OFFSET,
-)
-from .helpers import dt_local_to_aware, handle_evo_exception
+
+from .const import CONF_LOCATION_IDX, DOMAIN, GWS, TCS, UTC_OFFSET
+from .helpers import handle_evo_exception
+
+if TYPE_CHECKING:
+    from . import EvoSession
 
 _LOGGER = logging.getLogger(__name__.rpartition(".")[0])
 
 
 class EvoBroker:
-    """Container for evohome client and data."""
+    """Broker for evohome client broker."""
+
+    def __init__(self, sess: EvoSession) -> None:
+        """Initialize the evohome broker and its data structure."""
+
+        self._sess = sess
+        self.hass = sess.hass
+
+        assert isinstance(sess.client_v2, evo.EvohomeClient)  # mypy
+
+        self.client = sess.client_v2
+        self.client_v1 = sess.client_v1
+
+        self.loc_idx: int = None  # type: ignore[assignment]
+        self.loc: evo.Location = None  # type: ignore[assignment]
+
+        self.loc_utc_offset: timedelta = None  # type: ignore[assignment]
+        self.tcs: evo.ControlSystem = None  # type: ignore[assignment]
```

Comment:
These should use : int | None

Reply:
Yeah - it was either a single `type: ignore[assignment]` (for each attr) here, or multiple `assert attr is not None` later, so I chose the latter as the easier path.

Reply:
You can do this much more cleanly as the class level
```python
class EvoBroker:
    """Broker for evohome client broker."""

    loc_idx: int
    loc: evo.Location

    loc_utc_offset: timedelta
    tcs: evo.ControlSystem

    def __init__(self, sess: EvoSession) -> None:
        """Initialize the evohome broker and its data structure."""
```

Reply:
> You can do this much more cleanly as the class level

I've avoided this - I never knew what would happen if there was more than one instance of the class.